src/edu/csus/ecs/pc2/ui/ShadowCompareRunsPane.java

```diff
@@ -303,10 +361,372 @@ public void actionPerformed(ActionEvent arg0) {
+                boolean isClicsAcronym = CLICSJudgementType.isCLICSAcronym(remoteJudgement);
+                if (isClicsAcronym) {
+                    
+                    //get the CLICS acronym for the specified judgement 
+                    CLICS_JUDGEMENT_ACRONYM judgementAcronym = CLICSJudgementType.getCLICSAcronymFromElementName(remoteJudgement);
+
+                    // run is not pending in either system and the remote judgement is a valid CLICS judgement; 
+                    // attempt to update PC2 to match remote CCS judgement
+                    boolean success = updateRun(submissionId, judgementAcronym);
+                    
+                    if (success) {
+                        numRunsUpdated++;
+                    }
+                    
+                } else {
+                    
+                    //there is no known judgement matching the judgement string
+                    JOptionPane.showMessageDialog(this, "Cannot update submission " + submissionId + ": invalid remote judgement: " + remoteJudgement, 
+                                                        "Invalid judgement acronym", JOptionPane.WARNING_MESSAGE);
+                    log.log (Log.INFO, "Attempted to update pending submission: " + submissionId);
+
+                }
+            }
+        }
+        
+        //return the count of successfully updated runs
+        return numRunsUpdated;
+        
+    }
+
+    /**
+     * Invokes the PC2 server to update the status of the specified run to match the specified judgement.
+     * 
+     * @param submissionId the Id of the run to be updated.
+     * @param newJudgement the judgement which should be applied to the specified run.
+     * 
+     * @return true if the run was successfully updated; false if the run could not be updated for some reason 
+     *              (such as not being able to find the specified run, or the server failing to acknowledge a request
+     *              to edit/update the run within the server-timeout-limit).
+     */
+    protected boolean updateRun(Integer submissionId, CLICS_JUDGEMENT_ACRONYM newJudgement) {
+
+        log.log(Log.INFO, "Updating run " + submissionId + " to '" + newJudgement.name() + "'");
+        
+        //get all the runs
+        Run [] allRuns = getController().getContest().getRuns();
+        
+        //search for the desired run by Id
+        boolean found = false ;
+        Run targetRun = null;
+        for (Run run : allRuns) {
+            if (run.getNumber()==submissionId) {
+                targetRun = run;
+                found = true;
+                break;
+            }
+        }
+        
+        //if we didn't find the run, return failure
+        if (!found) {
+            
+            JOptionPane.showMessageDialog(this, "Failed to find submission " + submissionId, "Submission not found", JOptionPane.WARNING_MESSAGE);
+            log.log (Log.WARNING, "Failed to find run to be updated: " + submissionId);
```

Comment:
more log context update to:
"Failed to find run to be updated: " + submissionId +",  submission not found in run list "

Reply:
Done.